torch/csrc/autograd/engine.cpp

```diff
@@ -429,10 +429,11 @@ std::vector<Node*> get_current_graph_task_execution_order() {
     return {};
   }
 
-  // We could potentially check if there is only a single device here
-  // but explicitly require this context doesn't seem bad either
```

Comment:
Why change this behavior in this seemingly unrelated PR?

Reply:
Yeah this can be separated out